nerfstudio/viewer/viewer.py

```diff
@@ -462,11 +465,16 @@ def init_scene(
                 position=c2w[:3, 3] * VISER_NERFSTUDIO_SCALE_RATIO,
             )
 
-            @camera_handle.on_click
-            def _(event: viser.SceneNodePointerEvent[viser.CameraFrustumHandle]) -> None:
-                with event.client.atomic():
-                    event.client.camera.position = event.target.position
-                    event.client.camera.wxyz = event.target.wxyz
+            def create_on_click_callback(capture_idx):
+                def on_click_callback(event: viser.SceneNodePointerEvent[viser.CameraFrustumHandle]) -> None:
+                    with event.client.atomic():
+                        event.client.camera.position = event.target.position
+                        event.client.camera.wxyz = event.target.wxyz
+                        self.current_camera_idx = capture_idx
+
+                return on_click_callback
+
+            camera_handle.on_click(create_on_click_callback(idx))
```

Comment:
this also impacts nerfacto, right? could be good in its own PR?

Reply:
Basiclly this part will pass the camera idx in camera's metadata when renders a output. I guess it won't be a problem if the model doesn't use the camera metadata?